src/Microsoft.DotNet.ProjectModel/OutputPathCalculator.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public class OutputPathCalculator
```

Comment:
Shouldn't this class also be responsible for handling the intermediary output path?